modules/dnn/src/layers/layers_common.simd.hpp

```diff
@@ -101,9 +101,18 @@ void fastConv( const float* weights, size_t wstep, const float* bias,
 
         if( relu )
         {
-            r0 = relu[i];
-            r1 = relu[i+1];
-            r2 = relu[i+2];
+            if( i+1 >= outCn )
```

Comment:
@rockzhan, Actually, it should works correct because of extra elements:

https://github.com/opencv/opencv/blob/0f298a42032798d2d83017030621fcba885a3372/modules/dnn/src/layers/convolution_layer.cpp#L1007
https://github.com/opencv/opencv/blob/0f298a42032798d2d83017030621fcba885a3372/modules/dnn/src/layers/convolution_layer.cpp#L1009

Reply:
here the prelu weights add 2 extra elements at the end, but it should add 2 extra element to the front to handle 1 channel  situation.

Reply:
opencv/modules/dnn/src/layers/layers_common.simd.hpp
Line 97 in a75840d 
97   outptr2 = outptr1 = outptr0;

and here:
104   r0 = relu[i];
105   r1 = relu[i+1];
106   r2 = relu[i+2];

It will multiply the 3rd value and output.

Reply:
@rockzhan, May I ask you to add a test? Build a network with a single convolution layer followed by PReLU? You may find examples at https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_layers.cpp.

Reply:
sure, let me try

Reply:
@dkurt Hi, I add a test in the latest commit 9fb15dc3c55cbcfae2dae9afbde04a35368820d7.

Reply:
I create a caffe model with 2 layers  **[dwconv(group=1) + prelu]**, and manually set 
conv weights 3 channels as: 
> **weight[0]=-1 weight[1]=-2 weight[2]=-3 no_bias**,  <br>

set prelu weights as:  <br>

> **weight={1,2,3}**

Then create image size: **3x16x16** all pixel value is **1**
So the dwconv outout should be:
>1 1 1 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;                   -1 -1 -1
>1 1 1   **conv**    -1 -1 -1  **= -9** &nbsp; **,channel 1 output all -9**
>1 1 1 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;                   -1 -1 -1      
> &nbsp;&nbsp;&nbsp;**... channel 2 output all -18**
> &nbsp;&nbsp;&nbsp;**... channel 3 output all -27**  <br>

The expect prelu output should be:
> -9 x 1 &nbsp;&nbsp;= -9 &nbsp;&nbsp;  **,channel 1 output all -9**
> -18 x 2 = -36  **,channel 2 output all -36**
> -27 x 3 = -81  **,channel 3 output all -81**

But current opencv prelu output:
> -9 x 3 &nbsp;&nbsp;= -27 &nbsp;&nbsp;  **,channel 1 output all -27**
> -18 x 3 = -54  **,channel 2 output all -54**
> -27 x 3 = -81  **,channel 3 output all -81**

This will affect all models contain DepthwiseConv+Prelu block in OpenCV.

Reply:
Please make it in the same way as it is for bias values. Otherwise it doesn't work for number of output channels 6 and groups number 3, in example. I mean two `if` conditions `if( i+2 >= outCn )` and `if( i+1 >= outCn )`.

Reply:
I tested group = 2 or 3, the result is right. 
Because opencv make sure **outCn=1**, when use group convolution.
https://github.com/opencv/opencv/blob/b2cff44084b91c781d86b98bd2c99f90191a4ea1/modules/dnn/src/layers/convolution_layer.cpp#L513

Here

Reply:
and here:
https://github.com/opencv/opencv/blob/b2cff44084b91c781d86b98bd2c99f90191a4ea1/modules/dnn/src/layers/convolution_layer.cpp#L712-L713

Reply:
@rockzhan, try with number of output channels == 6 and group 3.

Reply:
yes I tested, output is right when use AVX2(number of output channels == 6 and group 3). and now I'm try to disable the optimizations.

Reply:
@rockzhan, `-DCV_DISABLE_OPTIMIZATION`

Reply:
> yes I tested, output is right when use AVX2(number of output channels == 6 and group 3). and now I'm try to disable the optimizations.

I just replaced 
```cpp
const int num_output = 3;
```
to
```cpp
const int num_output = 6;
```
and got
```
Expected: (normL1) <= (l1), actual: 8 vs 1e-05
Expected: (normInf) <= (lInf), actual: 32 vs 0.0001
```
Let's create a parametric test and check it on buildbot:
```cpp
typedef TestWithParam<int> Layer_Test_DWconv_Prelu;
TEST_P(Layer_Test_DWconv_Prelu, Accuracy)
{
    const int num_output = GetParam();
    ...
}
INSTANTIATE_TEST_CASE_P(/**/, Layer_Test_DWconv_Prelu, Values(3, 6));
```

Reply:
>Expected: (normL1) <= (l1), actual: 8 vs 1e-05
>Expected: (normInf) <= (lInf), actual: 32 vs 0.0001

This has 2 reason:
1. previous commit (before 26953c8) not fix without the optimizations, and now fixed.
2. In the Test, the target is fixed before, not flexible, and now fixed.

I will add a parametric test soon.